### DISCUSSION TITLE
Webpack is slower when loading minified files as input

### DISCUSSION THREAD

**Discussion by ling1726** [webpack/webpack#14922] [General]
# Bug report

**What is the current behavior?**

When loading large files, there is a significant time between webpack loading minified files vs unminified files **as input, not output**

**If the current behavior is a bug, please provide the steps to reproduce.**

Demo repo is here: https://github.com/ling1726/webpack-minified-slow

I tried to use the most basic webpack configuration possible

**What is the expected behavior?**

The code for 2 test cases is the identical apart from minification. I would expect that the times to bundle are fairly equal. However on my machine using the repro in the demo repo I get these results:

![image](https://user-images.githubusercontent.com/20744592/145027473-7b827be0-7031-4a24-9805-427ab0405083.png)

**Other relevant information:**
webpack version:  5.65.0
Node.js version: 14
Operating System: Windows with Linux subsystem
Additional tools: Some sample code I pulled from `@fluentui/react-icons@2.0.153-beta.4`

**Comment by alexander-akait** [comment]
What do you mean slower? Which metrics do you use?

**Reply by ling1726** [reply]
@alexander-akait  as I showed in the demo repo and the screenshot, I used the time to bundle. Both test cases had the same code and webpack config. Only difference was that the code was minified in one test

**Reply by alexander-akait** [reply]
Minification takes the time, it is expected, especial already minified

**Reply by ling1726** [reply]
I've updated the demo repo to disable the minimizer, and turned on the profiler. Here are the results:

![image](https://user-images.githubusercontent.com/20744592/145027373-af5dba8a-167a-47fb-a955-0a011cd4beec.png)

**Reply by ling1726** [reply]
> Minification takes the time, it is expected, especial already minified

To be clear, I don't mean that the output is minified, I meant that the input was minified

**Reply by alexander-akait** [reply]
hm, I see, I will look at this in near future

**Reply by alexander-akait** [reply]
The problem here https://github.com/webpack/webpack/blob/main/lib/javascript/JavascriptParser.js#L3725, acorn parses it very slow, that is very strange

**Reply by alexander-akait** [reply]
The problem in the `locations` option (if you set it to `false`, it will be faster, webpack throws an error, but parser will be faster as for prettified)

**Reply by alexander-akait** [reply]
Bug here https://github.com/acornjs/acorn/blob/master/acorn/src/tokenize.js#L97, try to comment
```
if (this.options.locations) {
      lineBreakG.lastIndex = start;
      var match;
      while ((match = lineBreakG.exec(this.input)) && match.index < this.pos) {
        ++this.curLine;
        this.lineStart = match.index + match[0].length;
      }
}
```

and no problems

**Reply by alexander-akait** [reply]
Now we need to do fix or create an issue, I will take it, thanks for the issue

**Reply by ling1726** [reply]
Can we reopen and track the original issue and close it when a fix is applied in acorn/webpack ?

**Reply by alexander-akait** [reply]
Reported https://github.com/acornjs/acorn/issues/1083

**Reply by alexander-akait** [reply]
@ling1726 We can do it, but we try to not keep issues which can't be solved in webpack/webpack ecosystem, they tend to accumulate and make noise, and keeping them will not speed up fixes, we use `^` https://github.com/webpack/webpack/blob/main/package.json#L13, so when acorn release a patch, you can update deps and the problem will be solved

**Reply by ling1726** [reply]
Sounds good, thanks!
